integer/src/lib.rs

```diff
@@ -665,6 +665,47 @@ impl_integer_for_usize!(u32, test_integer_u32);
 impl_integer_for_usize!(u64, test_integer_u64);
 impl_integer_for_usize!(usize, test_integer_usize);
 
+/// An iterator over binomial coefficients.
+pub struct BinomialCoeff<T> {
+    a: T,
+    n: T,
+    k: T,
+}
+
+impl<T> BinomialCoeff<T>
+    where T: Integer,
+{
+    /// For a given n, iterate over all binomial coefficients ((k, n - k), binomial(n, k)).
```

Comment:
It's odd to me that you include `(k, n - k)` in the output.  They're both redundant - `k` is available by converting from an `enumerate` index if you want it, or zipped with a `Range`, and `n - k` is easily derived from that.  I feel like this iterator should just stick to its interesting contribution, just the coefficients.